pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,189 @@
+package gke
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/auth/util"
+	client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/ref"
+	schema "github.com/rancher/rancher/pkg/schemas/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+)
+
+type Capabilities struct {
+	Credentials string `json:"credentials,omitempty"`
+	ProjectID   string `json:"projectId,omitempty"`
+	Zone        string `json:"zone,omitempty"`
+	Region      string `json:"region,omitempty"`
+}
+
+// GKE handler lists available resources in Google API
+type handler struct {
+	Action        string
+	schemas       *types.Schemas
+	secretsLister v1.SecretLister
+}
+
+func NewGKEHandler(scaledContext *config.ScaledContext) http.Handler {
+	return &handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	capa := &Capabilities{}
+
+	if credID := req.URL.Query().Get("cloudCredentialId"); credID != "" {
+		if err := h.getCloudCredential(req, capa, credID); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.InvalidBodyContent.Status, err.Error())
+			return
+		}
+	} else if req.Method == http.MethodPost {
+		if err := h.getCredentialsFromBody(writer, req, capa); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.InvalidBodyContent.Status, err.Error())
+			return
+		}
+	} else {
+		util.ReturnHTTPError(writer, req, httperror.Unauthorized.Status, "cannot access Google API without credentials to authenticate")
+	}
+
+	var serialized []byte
+	var errCode int
+	var err error
```

Comment:
nit: remove these since each case is handled in its own scope.

Reply:
this I would like keep it, when I removed it `goland` doesn't like it